### PR TITLE
Unified Build/Package/Test

### PR DESCRIPTION
**Pull request by compnerd**
This unifies all of the build infrastructure under CMake.  This also subsumes the testing infrastructure and the packaging infrastructure!

There is now a single canonical owner of all information related to building: CMake.

You can form Visual Studio projects, XCode projects, makefiles, Ninja based Linux builds, android builds from the CMake based build system.  This also includes installation.

CTest can be used to run the test suite (although, all the tests are currently treated a single entity).

CPack can be used to generate a tarball or a debian deb package or a RPM package.

Instructions are updated to reflect the various cases.

Overall, this represents a huge reduction in the size of items in the repository:
 29 files changed, 107 insertions(+), 8016 deletions(-)

Update the cmake dependency to something slightly more modern (although any 3.0+ version would be fine, I opt to take the same version as LLVM, clang, lld, lldb, swift as they support old enough environents that I dont think that it would be too onerous to request this version).

### CODE REVIEW

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -66,7 +42,24 @@ if(WIN32)
   if(CMAKE_C_COMPILER_ID MATCHES Clang)
     target_compile_options(libkqueue-test PRIVATE -Wno-unused-variable)
   endif()
+elseif(UNIX)
+  targeT_compile_options(libkqueue-test
```
Comment (mheily) [inline test/CMakeLists.txt]: Is this a typo? targeT instead of target..

**Comment by mheily** [comment]
I see a lot of autogenerated files when running "git status" after CMake. I think .gitignore should be updated to ignore the following:

	CMakeCache.txt
	CMakeFiles/
	CPackConfig.cmake
	CPackSourceConfig.cmake
	CTestTestfile.cmake
	Testing/
	_CPack_Packages/
	cmake_install.cmake
	install_manifest.txt
	kqueue-0.1.1-Linux.deb
	libkqueue.so
	test/CMakeCache.txt
	test/CMakeFiles/
	test/cmake_install.cmake

File: test/CMakeLists.txt (hunk 1 above)
Comment (compnerd) [inline test/CMakeLists.txt]: yeah, thats a typo, it should be `target_compile_options`.

**Comment by compnerd** [comment]
Sure, that sounds like a good list to ignore.  However, I think that we should encourage out-of-source builds :-).

**Pull request merged**